src/puppetlabs/puppetdb/anonymizer.clj

```diff
@@ -31,11 +32,10 @@
    (contains? event "old_value")
    (contains? event "resource_type")))
 
-(defn report?
+(pls/defn-validated report? :- report/full-report
   "Returns true if it looks like a report"
-  [report version]
-  ;; Utilise our existing report validation
-  (report/validate! version (keywordize-keys report)))
+  [report _]
+  (keywordize-keys report))
```

Comment:
sorry, what I meant was that the need for this function could be eliminated by just using schema on anonymize-report instead of pre/postchecks